### PR TITLE
feat: enforce upgrades on workspaces that exceed usage

### LINKED ISSUE #3505: Feature: Add blocking actions for users over the workspace thresholds
### Suggested solution

Currently, when users surpass the workspace thresholds, we show an upgrade banner at the top of the page for the workspace. This would change to preventing them from adding further repositories and contributors to their workspace or insight pages unless they upgrade.

Existing workspaces that are not upgraded would be affected also. The workspace upgrade limits are enforced through our API in each individual workspace.

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Blocks workspaces that 
  - Have exceeded the upgrade limits from adding additional workspace repositories
  - Have exceeded the upgrade limits from adding additional repositories to repository insights
  - Have exceeded the upgrade limits from adding additional contributors to contributor insights
  - Have exceeded the upgrade limits from adding additional collaborators to the workspace
- Updates copy about workspace thresholds

## Related Tickets & Documents

Closes #3505 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/42211/89850e08-59ad-4e97-a7c2-e145b7883c40

image

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

File: components/ListHeader/list-header.tsx [hunk 1]
```diff
@@ -78,7 +80,7 @@ const ListHeader = ({
         <Button onClick={() => handleCopyToClipboard()} className="px-6 py-2 bg-white " variant="text">
           <FiCopy className="mt-1 mr-2" /> Share
         </Button>
-        {isOwner && (
+        {isOwner && !overLimit && (
```
Comment (bdougie) [inline components/ListHeader/list-header.tsx:83]: We shouldn't hide the button if they are over the limit. Instead, show the modal when they click the button. 

Screenshot 2024-06-04 at 5 23 58 PM

**Review by bdougie — CHANGES_REQUESTED** [review]
Thanks for opening this. We want to be able to test out this change without creating a dark UX pattern. Perhaps I mentioned removing the button, we should instead leave it on the page and present the modal for upgrading.

File: components/ListHeader/list-header.tsx (hunk 1 above)
Comment (brandonroberts) [reply components/ListHeader/list-header.tsx]: Fixed

**Review by jpmcb — APPROVED** [review]
😎 💰

**Review by bdougie — APPROVED** [review]
@brandonroberts, just getting to review this. I think the "add members" should present the modal and not be disabled. That thought might present some extra UX decisions, so I am approving this, but would love to at minimum follow up on this in an issue. 

Screenshot 2024-06-06 at 2 28 48 PM

**Pull request merged**